lib/onelogin/ruby-saml/logoutresponse.rb

```diff
@@ -69,46 +104,211 @@ def status_code
         end
       end
 
+      # @return [String] Gets the StatusMessage from a SAML Response.
+      #
+      def status_message
+        @status_message ||= begin
+          node = REXML::XPath.first(document, "/p:LogoutResponse/p:Status/p:StatusMessage", { "p" => PROTOCOL, "a" => ASSERTION })
+          node.text if node
+        end
+      end
+
+      # Gets the expected current_url
+      # (Right now we read this url from the Sinle Logout Service of the Settings)
+      # TODO: Calculate the real current_url and use it.
+      # @return [String] The current url
+      #
+      def current_url
+        @current_url ||= begin
+          if settings && settings.single_logout_service_url
+            settings.single_logout_service_url
+          end
+        end
+      end
+
+      # Validates the Logout Response (calls several validation methods)
+      # If fails, the attribute errors will contains the reason for the invalidation.
+      # @param soft [Boolean] soft Enable or Disable the soft mode (In order to raise exceptions when the logout response is invalid or not)
+      # @param request_id [String|nil] request_id The ID of the Logout Request sent by this SP to the IdP (if was sent any)
+      # @param get_params [Hash] In order to validate the In value we need to provide the GET parameters
+      # @return [Boolean] True if the Logout Response is valid, otherwise False if soft=True
+      # @raise [ValidationError] if soft == false and validation fails
+      #
+      def validate(soft = true, request_id = nil, get_params = nil)
+        @errors = []
+        valid_state?(soft)                      &&
+        validate_success_status(soft)           &&
+        validate_structure(soft)                &&
+        valid_in_response_to?(soft, request_id) &&
+        validate_destination(soft)              &&
+        valid_issuer?(soft)                     &&
+        validate_signature(soft, get_params)
+      end
+
       private
 
+      # Validates that the Logout Response provided in the initialization is not empty, 
+      # also check that the setting and the IdP cert were also provided 
+      # If fails, the error is added to the errors array.
+      # @param soft [Boolean] soft Enable or Disable the soft mode (In order to raise exceptions when the logout response is invalid or not)
+      # @return [Boolean] True if the required info is found, otherwise False if soft=True
+      # @raise [ValidationError] if soft == false and validation fails
+      #
       def valid_state?(soft = true)
-        if response.empty?
-          return soft ? false : validation_error("Blank response")
+        if response.nil? or response.empty?
+          error_msg = "Blank Logout Response"
```

Comment:
the 

```ruby
error_msg = ...
@errors << error_msg
<ternary>
```
Construct is reused enough here that pulling it out into its own method would be worthwhile.

```ruby
def append_error(soft, error_msg)
  @errors << error_msg
   return soft ? false : validation_error(error_msg)
end
```